lib/portage/output.py

```diff
@@ -268,26 +268,67 @@ def nc_len(mystr):
 _legal_terms_re = re.compile(
     r"^(xterm|xterm-color|Eterm|aterm|rxvt|screen|kterm|rxvt-unicode|gnome|interix|tmux|st-256color|alacritty|konsole|foot)"
 )
-_disable_xtermTitle = None
-_max_xtermTitle_len = 253
+_xterm_title_supported = None
+_max_xterm_title_len = 253
+_title_init_seq = None
+_title_finish_seq = None
+_default_xterm_title = None
+
+
+def init_xterm_titles():
+    global _xterm_title_supported
+    global _title_init_seq
+    global _title_finish_seq
+
+    if _xterm_title_supported is not None:
+        # already initialized
+        return
+
+    _xterm_title_supported = False
+
+    if not sys.__stderr__.isatty() or "TERM" not in os.environ:
+        return
+
+    try:
+        # by default check if we can dynamically query the proper title
+        # setting sequence via terminfo
+        curses = init_curses(sys.stderr)
```

Comment:
We should be consistent in our usage of `sys.__stderr__` vs. `sys.stderr`, and `sys.__stderr__` is preferable because sometimes we override `sys.stderr` with a temporary `io.StringIO` instance.

Reply:
Maybe it's better if the fd argument defaults to `sys.__stderr__` in init_curses, so we'll only have to explicitly pass it in for an unusual case.

Reply:
Right, I've pushed a commit on top to address this.